aslam_cv_common/src/sensor.cc

```diff
@@ -0,0 +1,106 @@
+#include <aslam/common/sensor.h>
+
+namespace aslam {
+
+Sensor::Sensor(const SensorId& id) : id_(id) {
+  CHECK(id_.isValid());
+};
+
+Sensor::Sensor(const SensorId& id, const std::string& topic)
+    : id_(id), topic_(topic) {
+  CHECK(id_.isValid());
+};
+
+void Sensor::setId(const SensorId& id) {
+  id_ = id;
+  CHECK(id_.isValid());
+};
+
+void Sensor::setDescription(const std::string& description) {
+  description_ = description;
+}
+
+bool Sensor::isValid() const {
+  if (!id_.isValid()) {
+    LOG(ERROR) << "Invalid sensor id.";
+    return false;
+  }
+  return isValidImpl();
+}
+
+void Sensor::setRandom() {
+  generateId(&id_);
+  setRandomImpl();
+}
+
+bool Sensor::deserialize(const YAML::Node& sensor_node) {
+  if (!sensor_node.IsDefined() || sensor_node.IsNull()) {
+    LOG(ERROR) << "Invalid YAML node for sensor deserialization.";
+    return false;
+  }
+
+  if (!sensor_node.IsMap()) {
+    LOG(WARNING) << "Sensor YAML node must be a map.";
+    return false;
+  }
+
+  std::string id_as_string;
+  if (YAML::safeGet(
+          sensor_node, static_cast<std::string>(kYamlFieldNameId),
```

Comment:
No need for char, but this is the only reasonable way I think to to declare these string literals at compile time in the global space. And good alternatives like [string_view](https://en.cppreference.com/w/cpp/string/basic_string_view) are only in since C++17.